superx/routes/home.py

```diff
@@ -170,3 +174,20 @@ def city_search():
     """
     city_list = db.session.query(Branch.city).order_by(Branch.city).distinct().all()
     return render_template('city_list.html', city_list=city_list)
+
+
+def update_num_items():
+    """
+    renders number of items chosen for a product according to user's input
+    """
+    id_to_update = request.form.get('id')
+    num_items = request.form.get('num_items')
+
+    cart_list = session['cart']
+    for i in range(len(cart_list)):
+        if cart_list[i]['id'] == id_to_update:
+            cart_list[i]['num_items'] = num_items
+            break
+    session['cart'] = cart_list
```

Comment:
this is extra. cart_list is a reference to session['cart']  when either of them are updated, it is visible in the other.

Reply:
so actually after a few tests it appears that they do not update each other. 
without this line the code does not work appropriately, thus I did not change it

Reply:
that's weird. but I'm glad you checked it.